File: Leetcode/2697. Lexicographically Smallest Palindrome/solution.js
// https://leetcode.com/problems/lexicographically-smallest-palindrome/description/

const makeSmallestPalindrome = (s) => {
  let left = 0
  let right = s.length - 1

  let stringPal = s.split("")

  while(left < right) {
    const charLeft = s[left]
    const charRight = s[right]

    if(charLeft != charRight) {
      if(charLeft > charRight)
        stringPal[left] = charRight
      else{
        stringPal[right] = charLeft
      }
    }

    left ++
    right --
  }

  return stringPal.join("")
}

console.log(makeSmallestPalindrome("egcfe")) // "efcfe"

File: Leetcode/2697. Lexicographically Smallest Palindrome/solution.rb
def make_smallest_palindrome(s)
  left = 0
  right = s.size - 1

  stringPal = s

  while left < right
    charLeft = stringPal[left]
    charRight = stringPal[right]

    if left != right
      if charLeft > charRight
        stringPal[left] = charRight
        left += 1
        right -= 1
      else
        stringPal[right] = charLeft
        left += 1
        right -= 1
      end
    end
  end

  stringPal
end

puts make_smallest_palindrome("egcfe") "efcfe"
